medusa/show_updater.py

```diff
@@ -224,16 +227,23 @@ def clean_expired_seasons(self, show_list):
             if not [show for show in show_list if
                     show.indexer == row['indexer'] and
                     show.indexerid == row['indexer_id']]:
+
                 remove_row.append(row['indexer_update_id'])
                 remove_show.append(row['indexer_id'])
 
+        def get_rows_by_slice(seq, row_len):
+            """Simple pagination/slice method. Will transform [1,2,3,4] into [[1,2],[3,4]] when provided rowlen=2."""
+            for start in xrange(0, len(seq), row_len):
+                yield seq[start:start + row_len]
```

Comment:
Very nice! 

Just as an FYI, this could also be written as a gen exp. instead of a function:
```python
# Slice list into evenly sized chunks (e.g.  [1,2,3,4] --> [[1,2],[3,4]])
limit = 50  # max size for slice
rows_by_slice = (remove_row[i:i + n] 
                 for i in xrange(0, len(remove_list), limit))
```
Just keep in mind your function (and this version) don't work on iterables that don't have a length